cmd/check_spec.go

```diff
@@ -122,58 +92,419 @@ func newCheckSpec(ctx context.Context) (*checkSpec, error) {
+}
+
+func (cs *checkSpec) networkStatus(ctx context.Context) checkSpecOutput {
+	output := checkSpecOutput{
+		api: networkStatus,
+		validation: map[checkSpecRequirement]checkSpecStatus{
+			currentBlockID:   checkSpecSuccess,
+			currentBlockTime: checkSpecSuccess,
+			genesisBlockID:   checkSpecSuccess,
+			peers:            checkSpecSuccess,
+		},
+	}
+
 	res, err := cs.onlineFetcher.NetworkStatusRetry(ctx, Config.Network, nil)
 	if err != nil {
-		return fmt.Errorf("%w: unable to fetch network status", err.Err)
+		fmt.Printf("%v: unable to fetch network status\n", err.Err)
+		markAllValidationsFailed(output)
+		return output
 	}
 
-	if err := verifyBlockIdentifier(res.CurrentBlockIdentifier); err != nil {
-		return fmt.Errorf("%w", err)
+	// current_block_identifier is required
+	if err := validateBlockIdentifier(res.CurrentBlockIdentifier); err != nil {
+		fmt.Printf("%v\n", err)
+		setValidationStatus(output, currentBlockID, checkSpecFailure)
 	}
 
-	// TODO
-	return nil
+	// current_block_timestamp is required
+	if err := validateTimestamp(res.CurrentBlockTimestamp); err != nil {
+		fmt.Printf("%v\n", err)
+		setValidationStatus(output, currentBlockTime, checkSpecFailure)
+	}
+
+	// genesis_block_identifier is required
+	if err := validateBlockIdentifier(res.GenesisBlockIdentifier); err != nil {
+		fmt.Printf("%v\n", err)
+		setValidationStatus(output, genesisBlockID, checkSpecFailure)
+	}
+
+	// peers is required
+	if err := validatePeers(res.Peers); err != nil {
+		fmt.Printf("%v\n", err)
+		setValidationStatus(output, peers, checkSpecFailure)
+	}
+
+	fmt.Println("/network/status validated")
+	return output
 }
 
-func (cs *checkSpec) NetworkList(ctx context.Context, fetcher *fetcher.Fetcher) error {
+func (cs *checkSpec) networkList(ctx context.Context, fetcher *fetcher.Fetcher) checkSpecOutput {
+	output := checkSpecOutput{
+		api: networkList,
+		validation: map[checkSpecRequirement]checkSpecStatus{
+			networkIDs:      checkSpecSuccess,
+			offlineMode:     checkSpecSuccess,
+			staticNetworkID: checkSpecSuccess,
+		},
+	}
+
 	networks, err := fetcher.NetworkList(ctx, nil)
 	if err != nil {
-		return fmt.Errorf("%w: unable to fetch network list", err.Err)
+		fmt.Printf("%v: unable to fetch network list", err.Err)
+		markAllValidationsFailed(output)
+		return output
 	}
+
 	if len(networks.NetworkIdentifiers) == 0 {
-		return fmt.Errorf("network_identifiers are required")
+		fmt.Printf("network_identifiers is required")
+		setValidationStatus(output, networkIDs, checkSpecFailure)
 	}
+
 	for _, network := range networks.NetworkIdentifiers {
-		if network.Network == Config.Network.Network &&
-			network.Blockchain == Config.Network.Blockchain {
-			return nil
+		if network.Network != Config.Network.Network ||
+			network.Blockchain != Config.Network.Blockchain {
+			fmt.Printf("network_identifier in configuration file is not returned by /network/list")
+			setValidationStatus(output, staticNetworkID, checkSpecFailure)
 		}
 	}
```

Comment:
correct me if i'm wrong, the network ids should be different with each other? if under this assumption, this check seems always fail if ids are more than one.

Reply:
@irisZhangCB Could you give a bit more details? Didn't quite get "always fail if ids are more than one". 
This is just a syntax change (from "&&" to "||"), no logic modification.

Reply:
i'm thinking in the list of network ids, the test should pass whichever id listed in configuration file. but if put in a reverse way, i.e. check will fail if one of the ids is different than configuration file, it changes the logic and the output as well.

Reply:
Done.